system_metrics_collector/include/moving_average_statistics/types.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MOVING_AVERAGE_STATISTICS__TYPES_HPP_
+#define MOVING_AVERAGE_STATISTICS__TYPES_HPP_
+
+/**
+ *  A container for statistics results
+ */
+struct StatisticResults
+{
+  double average;
```

Comment:
I think it's better to initialize all the variables here.

Reply:
Good idea, will do.

Reply:
Consider keeping a 1:1 mapping between headers and types to facilitate the navigations. You could just call this `statistics_results.hpp`.

Regarding @prajakta-gokhale 's comment, the different is that keeping this w/o an inline initialization guarantee this type to be a POD struct, vs adding an inline initialization would make it non-POD.